packages/nest/src/shutdown.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable, OnApplicationShutdown } from '@nestjs/common';
+import { Provider } from '@openfeature/server-sdk';
+
+@Injectable()
+export class ShutdownService implements OnApplicationShutdown {
+  constructor(private readonly openFeatureProviders: Provider[]) {}
+
+  async onApplicationShutdown() {
+    const closePromises: Array<Promise<void>> = [];
+    for (const ofProvider of this.openFeatureProviders) {
+      if (ofProvider.onClose) {
+        closePromises.push(ofProvider.onClose());
+      }
+    }
+
+    await Promise.all(closePromises);
```

Comment:
```suggestion
    await OpenFeature.close()
```

I think we could just go for the close method of OpenFeature.
As the Application is shutdown anyways, we might just want to let OF manage the whole shutdown here.
This would also mean that we basically only have to create this service and not change anything about injecting the providers :)